File: backend/FragranceCommerce.Api/Models/Product.cs
namespace FragranceCommerce.Api.Models;

public class Product : BaseEntity
{
    public Guid VendorId { get; set; }
    public Vendor Vendor { get; set; } = null!;

    public Guid BrandId { get; set; }
    public Brand Brand { get; set; } = null!;

    public Guid CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public bool IsActive { get; set; } = true;

    public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();

    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();

    public ICollection<Review> Reviews { get; set; } = new List<Review>();
}

File: frontend/fragrance-commerce-web/src/types/product.ts
export type ProductGender = "Men" | "Women" | "Unisex";

export type ProductStatus = "Draft" | "Active" | "Deactivated";

export interface ProductImage {
    id: string;
    imageUrl: string;
    displayOrder: number;
    isPrimary: boolean;
}

export interface ProductVariantImage {
    id: string;
    imageUrl: string;
    displayOrder: number;
    isPrimary: boolean;
}

export interface ProductVariant {
    id: string;
    productId: string;
    variantName: string;
    sku: string;
    mrp: number;
    sellingPrice: number;
    costPrice: number;
    stockQuantity: number;
    isActive: boolean;
    images: ProductVariantImage[];
}

export interface Product {
    id: string;
    vendorId: string;
    vendorName?: string;

    brandId: string;
    brandName: string;

    categoryId: string;
    categoryName: string;

    gender: ProductGender;

    name: string;
    description?: string;
    status: ProductStatus;

    variants: ProductVariant[];
    images: ProductImage[];
}

export interface ProductSearchParams {
    search?: string;
    gender?: ProductGender;
    brandId?: string;
    categoryId?: string;
    minPrice?: number;
    maxPrice?: number;
    inStockOnly?: boolean;
    sortBy?: string;
    sortDirection?: string;
    pageNumber?: number;
    pageSize?: number;
}